storage/innobase/fil/fil0fil.cc

```diff
@@ -8420,7 +8420,9 @@ dberr_t Fil_shard::do_redo_io(const IORequest &type, const page_id_t &page_id,
   ut_a(byte_offset % OS_FILE_LOG_BLOCK_SIZE == 0);
 
   /* Set encryption information. */
+  mutex_acquire();
   fil_io_set_encryption(req_type, page_id, space);
+  mutex_release();
```

Comment:
I would avoid this if redo fil_reset_encryption() is never called.

Reply:
Although after reviewing I found a way for `fil_reset_encryption()` to be called for the redo log space, that involves stopping MySQL, modifying data on disk, and then starting MySQL again. That will cause much more apparent issues though, so I think it's safe to just assume this will never happen. Also, `Fil_shard::do_redo_io()` is never called on any other spaces other than the redo log space, so that path is also safe.

I've removed this part of the patch. Seems to be safe.